src/dc_tools.rs

```diff
@@ -640,19 +640,8 @@ impl FromStr for EmailAddress {
                 if local.is_empty() {
                     return err("empty string is not valid for local part");
                 }
-                if domain.len() <= 3 {
-                    return err("domain is too short");
-                }
-                let dot = domain.find('.');
-                match dot {
-                    None => {
-                        return err("invalid domain");
```

Comment:
The disadvantage is obviously that when the user types `alice@e` then we already ask whether they would like to create a new contact. Before, we waited until they typed `alice@example.o`, which is better UX; until they typed this last `o`, it said "Type email address above", clearly stating that the user is not done yet.

So, maybe only special-case `localhost` as a valid domain here?

Reply:
It may be useful to have plain hosts on some experimental mesh network, for example. Yggdrasil and similar networks provide security without the need for TLS. Not allowing to have `me@alice`, `me@bob` stands in the way of such experiments.

Maybe do not suggest creating uncommon contacts in the UI, but I think it should be possible if user wants to do it for some reason.

Reply:
If it's really a problem, it should be done only in `dc_may_be_valid_addr` or maybe some other better named function.

Reply:
@r10s what do you think? Should we allow email addresses whose domain does not contain a dot?

> Maybe do not suggest creating uncommon contacts in the UI, but I think it should be possible if user wants to do it for some reason.

You mean, make it possible to use `me@alice` when using the API but not when using one of our UIs?

> If it's really a problem, it should be done only in dc_may_be_valid_addr or maybe some other better named function.

Currently may_be_valid_addr is called in Contact::add_or_lookup() so that it would not be possible to chat with such a friend. But of course we could replace this call with `addr.parse::<EmailAddress>().is_ok()` and in only in `may_be_valid_addr` do stricter checks, so that the UIs don't treat them as valid email addresses.

This, in turn, would have the disadvantage that we might have contacts in our contact database - for whatever reason - for which `may_be_valid_addr` returns false. Not sure if that's a problem, either.

Reply:
i do not get it completely - does the change already affect `dc_may_be_valid_addr()` (and implicitly `dc_create_contact()` which calls `dc_may_be_valid_addr()`)?

if so, that would affect the UI, indeed. this is bad as fewer errors would be catched, as of `foo@gmailcom`. otoh, if it is a _real_ thing these mesh networks, it is probably better to allow them - but i would allow them everywhere then, so entering `foo@bar` should also be possible from ui.

Reply:
The change affects `dc_may_be_valid_addr`.

Reply:
correct, after applying the PR, as it currently is, DC would accept `foo@gmailcom` (OTOH, typos as `foo@gnail.com` probably happen far more often)

> but i would allow them everywhere then, so entering foo@bar should also be possible from ui.

yes, probably better not to distinguish between UI and API.

Reply:
+1 for `foo@gnail.com` :)

Reply:
@link2xt are there real tests with yggdrasil already happening? Otherwise adding localhost is probably good enough for now?